src/test/java/core/service/FruitStoreTest.java

```diff
@@ -0,0 +1,218 @@
+package core.service;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.exception.OperationHandlerException;
+import core.operationstrategy.OperationStrategyImpl;
+import core.transactions.BalanceOperationHandler;
+import core.transactions.OperationHandler;
+import core.transactions.PurchaseOperationHandler;
+import core.transactions.ReturnOperationHandler;
+import core.transactions.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FruitStoreTest {
+    private Map<OperationType, OperationHandler> strategyMap = new HashMap<>();
+    private OperationStrategyImpl strategy = new OperationStrategyImpl(strategyMap);
+    private FruitStore fruitStore;
+
+    @BeforeEach
+    public void setUp() {
+        strategyMap.put(OperationType.B, new BalanceOperationHandler());
+        strategyMap.put(OperationType.P, new PurchaseOperationHandler());
+        strategyMap.put(OperationType.S, new SupplyOperationHandler());
+        strategyMap.put(OperationType.R, new ReturnOperationHandler());
+
+        fruitStore = new FruitStore(strategy);
+        fruitStore.resetTransactions();
+    }
+
+    @Test
+    public void testProcessOperations_ValidData_ok() {
+        String data = "b,Apple,200" + System.lineSeparator()
```

Comment:
Please see common mistakes, 
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/69054906/768f35d5-a4ae-4536-84f7-ad56c9fe1f0e)

Reply:
not resolved

Reply:
Not resolved